Tools/AP_Periph/networking_passthru.cpp

```diff
@@ -0,0 +1,135 @@
+    // @Param: ENABLE
+    // @DisplayName: Enable Passthrough
+    // @Description: Enable Passthrough of any UART, Network, or CAN ports to any UART, Network, or CAN ports.
+    // @Values: 0:Disabled, 1:Enabled
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO_FLAGS("ENABLE", 1,  Networking_Periph::Passthru, enabled, 0, AP_PARAM_FLAG_ENABLE),
+
+    // @Param: EP1
+    // @DisplayName: Endpoint 1
+    // @Description: Passthrough Endpoint 1. This can be a serial port UART, a Network port, or a CAN port. The selected port will route to Endport 2.
+    // @Values: -1:Disabled, 0:Serial0(usually USB), 1:Serial1, 2:Serial2, 3:Serial3, 4:Serial4, 5:Serial5, 6:Serial6, 7:Serial7, 8:Serial8, 9:Serial9, 21:Network Port1, 22:Network Port2, 23:Network Port3, 24:Network Port4, 25:Network Port5, 26:Network Port6, 27:Network Port7, 28:Network Port8, 29:Network Port9, 41:CAN1 Port1, 42:CAN1 Port2, 43:CAN1 Port3, 44:CAN1 Port4, 45:CAN1 Port5, 46:CAN1 Port6, 47:CAN1 Port7, 48:CAN1 Port8, 49:CAN1 Port9, 51:CAN2 Port1, 52:CAN2 Port2, 53:CAN2 Port3, 54:CAN2 Port4, 55:CAN2 Port5, 56:CAN2 Port6, 57:CAN2 Port7, 58:CAN2 Port8, 59:CAN2 Port9
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO("EP1", 2,  Networking_Periph::Passthru, ep1, -1),
+
+    // @Param: EP2
+    // @DisplayName: Endpoint 2
+    // @Description: Passthrough Endpoint 2. This can be a serial port UART, a Network port, or a CAN port. The selected port will route to Endport 1.
+    // @CopyFieldsFrom: NET_PASS1_EP1
+    AP_GROUPINFO("EP2", 3,  Networking_Periph::Passthru, ep2, -1),
+
+    // @Param: BAUD1
+    // @DisplayName: Endpoint 1 Baud Rate
+    // @Description: The baud rate used for Endpoint 1. Only applies to serial ports.
+    // @CopyFieldsFrom: SERIAL1_BAUD
+    AP_GROUPINFO("BAUD1", 4,  Networking_Periph::Passthru, baud1, 115200),
+
+    // @Param: BAUD2
+    // @DisplayName: Endpoint 2 Baud Rate
+    // @Description: The baud rate used for Endpoint 2. Only applies to serial ports.
+    // @CopyFieldsFrom: SERIAL1_BAUD
+    AP_GROUPINFO("BAUD2", 5,  Networking_Periph::Passthru, baud2, 115200),
+
+    // @Param: OPT1
+    // @DisplayName: Serial Port Options EP1
+    // @Description: Control over UART options for Endpoint 1. Only applies to serial ports.
+    // @CopyFieldsFrom: SERIAL1_OPTIONS
+    AP_GROUPINFO("OPT1", 6,  Networking_Periph::Passthru, options1, 0),
+
+    // @Param: OPT2
+    // @DisplayName: Serial Port Options EP2
+    // @Description: Control over UART options for Endpoint 2. Only applies to serial ports.
+    // @CopyFieldsFrom: SERIAL1_OPTIONS
+    AP_GROUPINFO("OPT2", 7,  Networking_Periph::Passthru, options2, 0),
+
+    AP_GROUPEND
+};
+
+void Networking_Periph::Passthru::init()
+{
+    if (enabled == 0) {
+        // Feature is disabled
+        return;
+    }
+
+    if (port1 != nullptr || port2 != nullptr) {
+        // Endpoints have already been initilized
```

Comment:
small error, initialized

Reply:
Thanks. Changed to "// The ports have already been initialized, nothing to do."